src/wutil.cpp

```diff
@@ -564,7 +564,7 @@ file_id_t file_id_t::file_id_from_stat(const struct stat *buf)
 #if defined(__APPLE__) || defined(__DragonFly__) || defined(__FreeBSD__) ||  defined(__OpenBSD__) || defined(__NetBSD__)
     result.generation = buf->st_gen;
 #else
-    result.generation = 0;
+    result.generation = NULL;
```

Comment:
This is wrong because the generation member of the struct is a uint32_t not a pointer. But as I've hinted in my other comments regarding this change I think we should just drop the generation member.